File: program/program/convert-hexadecimal-to-decimal/convert_hexadecimal_to_decimal.go
package main

import (
	"fmt"
	"math"
)

func hexadecimalToDecimal(hexadecimalNumber string) int {
	var decimalNumber int
	var power int = 0

	for i := len(hexadecimalNumber) - 1; i >= 0; i-- {
		digit := hexadecimalNumber[i]

		if digit >= '0' && digit <= '9' {
			decimalNumber += int(digit-'0') * int(math.Pow(16, float64(power)))
		} else if digit >= 'A' && digit <= 'F' {
			decimalNumber += int(digit-'A'+10) * int(math.Pow(16, float64(power)))
		}

		power++
	}

	return decimalNumber
}

func main() {
	hexadecimalNumber := "F"
	decimalNumber := hexadecimalToDecimal(hexadecimalNumber)
	fmt.Println("Hexadecimal number:", hexadecimalNumber)
	fmt.Println("Decimal number:", decimalNumber)
}

File: program/program/convert-hexadecimal-to-decimal/ConvertHexadecimalToDecimal.c
// C program to convert hexadecimal to decimal

#include <math.h>
#include <stdio.h>
#include <string.h>
int main()
{
	char hexdecnumber[17] = "2D";
	int decimalnumber, place;
	int i = 0, val, len;
	decimalnumber = 0;

	// finding the length of hexadecnumber
	len = strlen(hexdecnumber);
	len--;

	// while loop executes the statements until the
	// condition is false
	while (hexdecnumber[i] != '\0') {

		// finding the equivalent decimal digit for each
		// hexa decimal digit
		if (hexdecnumber[i] >= '0'
			&& hexdecnumber[i] <= '9') {
			val = hexdecnumber[i] - 48;
		}
		else if (hexdecnumber[i] >= 'a'
				&& hexdecnumber[i] <= 'f') {
			val = hexdecnumber[i] - 97 + 10;
		}
		else if (hexdecnumber[i] >= 'A'
				&& hexdecnumber[i] <= 'F') {
			val = hexdecnumber[i] - 65 + 10;
		}
		// final decimal number
		decimalnumber += val * pow(16, len);
		len--;
		i++;
	}
	// printing the result
	printf("Hexadecimal number = %s\n", hexdecnumber);
	printf("Decimal number = %d", decimalnumber);
	return 0;
}
